common/city.cpp

```diff
@@ -2247,6 +2247,9 @@ void add_specialist_output(
     // This is more than just an optimization. For governors that forbid
     // specialists, the cache may not be filled.
     if (count > 0) {
+      // If there is a cache it must not be empty.
+      fc_assert(!pcsoutputs || !pcsoutputs->empty());
```

Comment:
`fc_` asserts aren't fatal by default. Use `fc_assert_action(condition, continue)` to skip this specialist, which is probably an ok fallback behavior:
```suggestion
      fc_assert_action(!pcsoutputs || !pcsoutputs->empty(), continue);
```

Reply:
I implemented another solution, falling back to the old behaviour by setting the cache to `nullptr` if the assertion triggers. This makes sure, that the game will behave correctly. What do you think?